### PR TITLE
Upgrade Terraform, Remove Terragrunt

### PR DESCRIPTION
**Pull request by grahamlyons**
The latest version of Terraform has support for managing remote state and providing locks using S3 and DynamoDB, respectively. We'd like to use the latest version of Terraform so we can get the latest features and bug-fixes, and it allows us to drop the Terragrunt dependency.

This makes the required changes to stop writing out Terragrunt config and configure Terraform instead.

### CODE REVIEW

File: cdflow_commands/plugins/ecs.py [hunk 1]
```diff
@@ -97,9 +98,15 @@ def _deploy_factory():
         )
         s3_bucket_factory = S3BucketFactory(boto_session, account_id)
         s3_bucket = s3_bucket_factory.get_bucket_name()
-        write_terragrunt_config(
-            metadata.aws_region, s3_bucket, environment_name, component_name
+
+        lock_table_factory = LockTableFactory(boto_session)
```
Comment (bart613) [inline cdflow_commands/plugins/ecs.py]: I've got two comments around this bit;

1) `lock_table_factory` as variable name sounds weird to me, would it not be more natural to just call it `terraform_lock`?  The Class name sounds fine as there's also a good split between LockTable and S3 bucket for state.
2) Having that in `build_ecs_plugin` means we need to duplicate this in `infrastructure` plugin, too?  Perhaps it'd be better to acquire a lock before those things are happening?

File: cdflow_commands/state.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+from shutil import move
+from subprocess import check_call
+from tempfile import NamedTemporaryFile
+from textwrap import dedent
+
+from botocore.exceptions import ClientError
+from cdflow_commands.exceptions import CDFlowError
+from cdflow_commands.logger import logger
+
+TAG_NAME = 'is-cdflow-tfstate-bucket'
```
Comment (bart613) [inline cdflow_commands/state.py]: What's the reason behind `is-` prefix?

Comment (grahamlyons) [inline cdflow_commands/state.py]: 🤷‍♀️  Dunno mate. It's only got my name against it because I renamed the file.

Comment (bart613) [inline cdflow_commands/state.py]: I think `cdflow-tfstate-bucket` would be a better suited Tag Name.

File: cdflow_commands/plugins/ecs.py (hunk 1 above)
Comment (grahamlyons) [inline cdflow_commands/plugins/ecs.py]: This doesn't acquire the lock it just sets up the table which Terraform then uses.

We do duplicate it, which is annoying. There's other bits duplicated between them which we should look at consolidating.

Comment (bart613) [inline cdflow_commands/plugins/ecs.py]: Okay , fair enough.

**Comment by grahamlyons** [comment]
To address your points:
1. No, if we can't create the table then we fail. I did start writing logic to create a table with a different name etc. but I figured it was a bit too complex - try to get the table, if it doesn't exist then create it. Fail in all other cases.
2. No, this is still the same situation where Terraform controls the lock and releases it once it's done it's thing.

**Comment by bart613** [comment]
👍

**Pull request merged**